servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/AsyncContext.java

```diff
@@ -104,11 +226,30 @@ public static AsyncContextMap current() {
      * @throws UnsupportedOperationException if this method is not supported by the underlying {@link AsyncContextMap}
      * implementation.
      * @see AsyncContextMap#putAll(Map)
+     * @deprecated Use {@link #putAllFromMap(Map)}
      */
-    public static void putAll(Map<AsyncContextMap.Key<?>, Object> map) {
+    @Deprecated
+    public static void putAll(final Map<AsyncContextMap.Key<?>, Object> map) {
         current().putAll(map);
     }
 
+    /**
+     * Convenience method for to put all the key-value pairs into the current context.
+     *
+     * @param map contains the key-value pairs that will be added.
+     * @throws ConcurrentModificationException done on a best effort basis if {@code entries} is detected to be modified
+     * while attempting to put all entries.
+     * @throws NullPointerException (optional behavior) if any of the {@code map} entries has a {@code null} {@code key}
+     * or {@code value} and the underlying {@link ContextMap} implementation doesn't support {@code null} keys or
+     * values.
+     * @throws UnsupportedOperationException if this method is not supported by the underlying {@link ContextMap}
+     * implementation.
+     * @see ContextMap#putAll(Map)
+     */
+    public static void putAllFromMap(final Map<ContextMap.Key<?>, Object> map) {
```

Comment:
Java complains if I have two methods that take a `Map` of something, because parameters will be replaced at compilation time. Therefore, had to come up with a different name. We can add a `putAll` alias (and deprecate `putAllFromMap`) in 0.42. I will add a comment to clarify.